src/app/games/yinsh/tests/yinsh.component.spec.ts

```diff
@@ -762,5 +768,87 @@ describe('YinshComponent', () => {
             testUtils.expectElementNotToExist('#ring_3_2');
             testUtils.expectElementNotToHaveClass('#pieceGroup_3_2', 'semi-transparent');
         }));
+        it('should show indicator when selecting your ring', fakeAsync(async() => {
+            // Given an initial board on which a ring are all put
+            const board: Table<YinshPiece> = [
+                [N, N, N, N, N, N, _, _, _, _, N],
+                [N, N, N, N, _, _, _, _, _, _, _],
+                [N, N, N, A, A, B, _, _, _, _, _],
+                [N, N, _, a, _, _, _, _, _, _, _],
+                [N, _, _, a, _, _, _, _, _, _, _],
+                [N, _, _, _, _, _, _, _, _, _, N],
+                [_, _, _, _, _, _, _, _, _, _, N],
+                [_, _, _, _, _, _, _, _, _, N, N],
+                [_, _, _, _, _, _, _, _, N, N, N],
+                [_, _, _, _, _, _, _, N, N, N, N],
+                [N, _, _, _, _, N, N, N, N, N, N],
+            ];
+            const state: YinshState = new YinshState(board, [0, 0], 10);
+            testUtils.setupState(state);
+
+            // When clicking on the ring
+            await testUtils.expectClickSuccess('#click_3_2');
+
+            // Then there should be indicators
+            testUtils.expectElementToExist('#indicator_4_1'); // The one in the up-right diagonal
+            testUtils.expectElementToExist('#indicator_2_3'); // The two in the down-left diagonal
+            testUtils.expectElementToExist('#indicator_1_4');
+        }));
+        it('should cancel move attempt chosen ring when clicking on it again', fakeAsync(async() => {
+            // Given an initial board on which a ring has been clicked (hence, helper displayed)
```

Comment:
helper displayed ? wat?

Reply:
well, the small rectangle, the helpers ? ((hence, the indicators are displayed) now !